test/cmd/annotations.sh

```diff
@@ -13,16 +13,16 @@ trap os::test::junit::reconcile_output EXIT
 os::test::junit::declare_suite_start "cmd/annotate"
 # This test validates empty values in key-value pairs set by the annotate command
 os::cmd::expect_success_and_text 'oc process -f examples/zookeeper/template.json | oc apply -f -' 'pod "zookeeper-1" created'
-os::cmd::expect_success_and_text 'oc annotate pod zookeeper-1 node-selector=""' 'pod "zookeeper-1" annotated'
-os::cmd::expect_success_and_text 'oc get pod zookeeper-1 --template="{{.metadata.annotations}}"' 'node-selector:'
+os::cmd::expect_success_and_text 'oc annotate pod zookeeper-1 nodeselector=""' 'pod "zookeeper-1" annotated'
+os::cmd::expect_success_and_text 'oc get pod zookeeper-1 --template="{{.metadata.annotations.nodeselector}}"' ''
 
 echo "annotate: ok"
 os::test::junit::declare_suite_end
 
 os::test::junit::declare_suite_start "cmd/label"
 # This test validates empty values in key-value pairs set by the label command
 os::cmd::expect_success_and_text 'oc label pod zookeeper-1 label2=""' 'pod "zookeeper-1" labeled'
-os::cmd::expect_success_and_text 'oc get pod zookeeper-1 --template="{{.metadata.labels}}"' 'label2\: '
+os::cmd::expect_success_and_text 'oc get pod zookeeper-1 --template="{{.metadata.labels.label2}}"' ''
```

Comment:
if you want to ensure the empty value, you need to pin to start and end: '^$'
if the label is completely missing, you'll get '<no value>' from the template output and '' will match that since it does substring regex match